internal/utils/nodeGroup.go

```diff
@@ -23,90 +29,178 @@ func CalculateNodeGroup(ctx context.Context, nodes v1.NodeList, config danav1alp
+func UpdateRootSubnamespace(ctx context.Context, rootResources v1.ResourceList, rootSubnamespace danav1alpha1.SubnamespacesRoots, logger logr.Logger, client client.Client) error {
+	rootRQ, err := GetRootQuota(client, ctx, rootSubnamespace.RootNamespace)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Error getting the %s resourceQuota", rootSubnamespace.RootNamespace))
+		return err
+	}
+	rootRQ.Spec.Hard = rootResources
+	logger.Info(fmt.Sprintf("Updating RootSubnamespace %s with new resources", rootSubnamespace.RootNamespace))
+	if err := client.Update(ctx, &rootRQ); err != nil {
+		logger.Error(err, fmt.Sprintf("Error updating rootSubnamespace %s", rootSubnamespace.RootNamespace))
+		return err
+	}
+	return nil
+}
+
+// UpdateRootSubnamespace updates the secondaryRoots in the cluster with the new quantity of resources.
+// It takes slice of Subnamespaces that was updated in memory and does API requests to commit the update.
+func UpdateProcessedSecondaryRoots(ctx context.Context, processedSecondaryRoots []danav1.Subnamespace, logger logr.Logger, client client.Client) error {
+	for _, sns := range processedSecondaryRoots {
+		logger.Info(fmt.Sprintf("Updating secondaryRoot %s with new resources", sns.Name))
+		if err := client.Update(ctx, &sns); err != nil {
+			logger.Error(err, fmt.Sprintf("Error updating secondaryRoot %s", sns.Name))
+			return err
+		}
 	}
+	return nil
 }
 
-func ProcessSecondaryRoot(ctx context.Context, r client.Client, secondaryRoot danav1alpha1.NodeGroup, config *danav1alpha1.NodeQuotaConfig, rootSubnamespace string) (error, v1.ResourceList) {
+// isReservedResourceExpired checks if a reserved created more than X hours ago, defined by the user in the config CRD.
+func isReservedResourceExpired(reservedResources danav1alpha1.ReservedResources, config danav1alpha1.NodeQuotaConfig) bool {
+	return hoursPassedSinceDate(reservedResources.Timestamp) >= config.Spec.ReservedHoursToLive
+}
+
+// setReservedToConfig sets the reserved resources for a node group in the NodeQuotaConfig.
+// It takes the resource debt (v1.ResourceList) to set, the node group name, the NodeQuotaConfig to modify, and a logger for logging informational messages.
+func setReservedToConfig(debt v1.ResourceList, nodeGroupName string, config *danav1alpha1.NodeQuotaConfig, logr logr.Logger) {
+	if doesReservedResourceExist(*config, nodeGroupName) {
+		reservedResources := getReservedResourcesByGroup(nodeGroupName, *config)
+		reservedResources.Resources = debt
+		removeReservedFromConfig(nodeGroupName, config)
+		config.Status.ReservedResources = append(config.Status.ReservedResources, reservedResources)
+		return
+	}
+	config.Status.ReservedResources = append(config.Status.ReservedResources, danav1alpha1.ReservedResources{
+		NodeGroup: nodeGroupName,
+		Resources: debt,
+		Timestamp: metav1.Now(),
+	})
+	logr.Info(fmt.Sprintf("Added ReservedResources to nodeGroup %s", nodeGroupName))
+}
```

Comment:
Not sure I understand the code here. Can't this function be re-written as something simpler like:

```
func setReservedToConfig(debt v1.ResourceList, nodeGroupName string, config *danav1alpha1.NodeQuotaConfig, logr logr.Logger) {
	if doesReservedResourceExist(*config, nodeGroupName) {
		removeReservedFromConfig(nodeGroupName, config)
	}
	config.Status.ReservedResources = append(config.Status.ReservedResources, danav1alpha1.ReservedResources{
		NodeGroup: nodeGroupName,
		Resources: debt,
		Timestamp: metav1.Now(),
	})
	logr.Info(fmt.Sprintf("Added ReservedResources to nodeGroup %s", nodeGroupName))
}
```

Reply:
Also should probably rename the function from `setReservedToConfig` to `addReservedResourcesToConfigStatus` since this is what it actually does

Reply:
I am doing this so the `timestamp` won't get overridden with `metav1.now()` and will cause endless updates